multiaddr/protocols.py

```diff
@@ -175,46 +179,41 @@ def read_varint_code(buf):
 
 
 def add_protocol(proto):
-    if proto.name in _names_to_protocols:
-        raise ValueError("protocol by the name %s already exists" % proto.name)
+	if proto.name in _names_to_protocols:
+		raise exceptions.ProtocolExistsError(proto, "name")
```

Comment:
It seems we are mixing tabs here, whereas the original code is indented with spaces.